File: app/src/main/java/forpdateam/ru/forpda/views/messagepanel/SimpleInstruction.java
package forpdateam.ru.forpda.views.messagepanel;

import android.content.Context;
import android.widget.Button;
import android.widget.ScrollView;
import android.widget.TextView;

import forpdateam.ru.forpda.R;

/**
 * Created by radiationx on 26.05.17.
 */

public class SimpleInstruction extends ScrollView {
    private TextView messageView;
    private Button closeButton;
    private OnClickListener listener;

    public SimpleInstruction(Context context) {
        super(context);
        addView(inflate(context, R.layout.message_panel_instruction, null));
        setFillViewport(true);
        messageView = (TextView) findViewById(R.id.instruction_message);
        closeButton = (Button) findViewById(R.id.instruction_close_button);
        closeButton.setOnClickListener((v) -> {
            this.setVisibility(GONE);
            if (listener != null) {
                listener.onClick(v);
            }
        });
    }

    public void setText(String text) {
        messageView.setText(text);
    }

    public void setOnCloseClick(OnClickListener listener) {
        this.listener = listener;
    }
}

File: app/src/main/java/forpdateam/ru/forpda/ui/views/messagepanel/SimpleInstruction.kt
package forpdateam.ru.forpda.ui.views.messagepanel

import android.content.Context
import android.view.View
import android.widget.Button
import android.widget.ScrollView
import android.widget.TextView
import forpdateam.ru.forpda.R

/**
 * Created by radiationx on 26.05.17.
 */
class SimpleInstruction(context: Context?) : ScrollView(context) {
    private val messageView: TextView
    private val closeButton: Button
    private var listener: OnClickListener? = null

    init {
        addView(inflate(context, R.layout.message_panel_instruction, null))
        isFillViewport = true
        messageView = findViewById(R.id.instruction_message)
        closeButton = findViewById(R.id.instruction_close_button)
        closeButton.setOnClickListener { v: View? ->
            this.visibility =
                GONE
            if (listener != null) {
                listener!!.onClick(v)
            }
        }
    }

    fun setText(text: String?) {
        messageView.text = text
    }

    fun setOnCloseClick(listener: OnClickListener?) {
        this.listener = listener
    }
}
